cpp/src/arrow/sparse_tensor.cc

```diff
@@ -503,72 +515,111 @@ Status MakeTensorFromSparseTensor(MemoryPool* pool, const SparseTensor* sparse_t
 // ----------------------------------------------------------------------
 // SparseCOOIndex
 
-Status SparseCOOIndex::Make(std::shared_ptr<DataType> indices_type,
-                            const std::vector<int64_t>& indices_shape,
-                            const std::vector<int64_t>& indices_strides,
-                            std::shared_ptr<Buffer> indices_data,
-                            std::shared_ptr<SparseCOOIndex>* out) {
-  *out = std::make_shared<SparseCOOIndex>(std::make_shared<Tensor>(
-      indices_type, indices_data, indices_shape, indices_strides));
+namespace {
+
+inline Status CheckSparseCOOIndexValidity(const std::shared_ptr<DataType>& type,
+                                          const std::vector<int64_t>& shape,
+                                          const std::vector<int64_t>& strides) {
+  if (!is_integer(type->id())) {
+    return Status::Invalid("Type of SparseCOOIndex indices must be integer");
```

Comment:
Can it be both signed or unsigned? Or do you have to check for that as well?

Reply:
It can be both signed or unsigned.